nf_core/utils.py

```diff
@@ -1009,8 +1010,12 @@ def get_repo_releases_branches(pipeline, wfs):
         ):
             wf_branches[branch["name"]] = branch["commit"]["sha"]
 
+        # Get commit information from github api
+        commit_response = gh_api.safe_get(f"https://api.github.com/repos/{pipeline}/commits?sha={branch['name']}")
+        for commit in commit_response.json():
+            wf_commits.append(commit["sha"])
```

Comment:
I am a bit scared of this - is it really getting every single commit from a repo? In some repos that could be a huge response.

I wonder if we move this into a separate function call and only do it if we need to - that is, once we're sure that the user is passing a commit and is not going to use a branch or a tag?

I suspect that the number of people wanting exact commits to be a tiny proportion, so I don't want to slow down / risk crashes for the majority.